internal/hardware/validator.go

```diff
@@ -63,31 +65,49 @@ func isNvme(name string) bool {
 	return strings.HasPrefix(name, "nvme")
 }
 
-func ListValidDisks(inventory *models.Inventory, minSizeRequiredInBytes int64) []*models.Disk {
-	var disks []*models.Disk
-	for _, disk := range inventory.Disks {
-		if disk.SizeBytes >= minSizeRequiredInBytes && funk.ContainsString([]string{"HDD", "SSD"}, disk.DriveType) {
-			disks = append(disks, disk)
-		}
+// DiskIsEligible checks if a disk is relevant for installation by testing
+// it against a list of predicates. Also returns all the reasons the disk
+// was found to be not eligible
+func (v *validator) DiskIsEligible(disk *models.Disk) (relevant bool, notEligibleReasons []string) {
+	if minSizeBytes := gbToBytes(v.MinDiskSizeGb); disk.SizeBytes < minSizeBytes {
+		notEligibleReasons = append(notEligibleReasons,
+			fmt.Sprintf("Disk is too small (disk only has %d bytes, but %d bytes are required)", disk.SizeBytes, minSizeBytes))
```

Comment:
Let's show the user GB, not bytes (note that for disks you should use GB not GiB, i.e., you should use base 10)

Reply:
Fixed with [go-humanize](https://github.com/dustin/go-humanize), so even small sizes don't look weird.